firebase-firestore/src/main/java/com/google/firebase/firestore/core/Filter.java

```diff
@@ -15,11 +15,19 @@
 package com.google.firebase.firestore.core;
 
 import com.google.firebase.firestore.model.Document;
+import com.google.firebase.firestore.model.FieldPath;
+import java.util.List;
 
 public abstract class Filter {
   /** Returns true if a document matches the filter. */
   public abstract boolean matches(Document doc);
 
   /** A unique ID identifying the filter; used when serializing queries. */
   public abstract String getCanonicalId();
+
+  /** Returns a list of all field filters that are contained within this filter */
+  public abstract List<FieldFilter> getAllFieldFilters();
+
+  /** Returns the field of the first filter that's an inequality, or null if none. */
+  public abstract FieldPath inequalityField();
```

Comment:
Please prefix with "get" (and possibly future-proof it by making it `getFirstInequalityField`)